### ISSUE TITLE
Forwarded `iterator()` of `Array` is not inlined in `for`

### ISSUE THREAD

**Issue by SlavaRa** [HaxeFoundation/haxe#5981]
Haxe code for example:
```haxe
package;

class Main {
	static function main() {
		var a:AArray<Int> = [1, 2, 3, 4, 5];
		for(it in a) trace(it);
	}
}

@:forward(iterator)
abstract AArray<T>(Array<T>) from Array<T> to Array<T> {
	public function new() this = [];
}
```

Result js code:
```js
// Generated by Haxe 3.4.0
(function () { "use strict";
var HxOverrides = function() { };
HxOverrides.iter = function(a) {
	return { cur : 0, arr : a, hasNext : function() {
		return this.cur < this.arr.length;
	}, next : function() {
		return this.arr[this.cur++];
	}};
};
var Main = function() { };
Main.main = function() {
	var it = HxOverrides.iter([1,2,3,4,5]);
	while(it.hasNext()) console.log(it.next());
};
var _$Main_AArray_$Impl_$ = {};
_$Main_AArray_$Impl_$._new = function() {
	return [];
};
Main.main();
})();
```

Haxe code without `abstract`:
```haxe
package;

class Main {
	static function main() {
		var a:Array<Int> = [1, 2, 3, 4, 5];
		for(it in a) trace(it);
	}
}
```

Result js code:
```js
// Generated by Haxe 3.4.0
(function () { "use strict";
var Main = function() { };
Main.main = function() {
	var a = [1,2,3,4,5];
	var _g = 0;
	while(_g < a.length) {
		var it = a[_g];
		++_g;
		console.log(it);
	}
};
Main.main();
})();
```

**Comment by nadako** [comment]
We need to introduce `ArrayIterator<T>` with inline methods and return that from all implementations of `Array<T>`, I think that should fix it.

**Comment by nadako** [comment]
I'll work on it soon(-ish).

**Comment by nadako** [comment]
Meanwhile, this should work nicely:

```haxe
abstract A(Array<Int>) from Array<Int> {
	var length(get,never):Int;
	inline function get_length() return this.length;
	@:arrayAccess inline function get(idx) return this[idx];
}

class Main {
	static function main() {
		var a:A = [1,2,3];
		for (v in a) {
			trace(a);
		}
	}
}
```

generating:
```js
Main.main = function() {
	var a = [1,2,3];
	var _g = 0;
	while(_g < a.length) {
		++_g;
		console.log("Main.hx:63:",a);
	}
};
```

**Comment by RealyUniqueName** [comment]
We already implemented custom iterators. But this solution looks better. Thanks.

**Comment by Simn** [comment]
This isn't about abstracts but about using the `iterator` field. You get the same output with this code:

```haxe
class Main {
	static function main() {
		var a = [1, 2, 3, 4, 5];
		for(it in a.iterator()) trace(it);
	}
}
```

It might be worthwhile to treat this as a special case, though @nadako's approach would also fix it.

**Comment by RealyUniqueName** [comment]
Fixed by #8344
